driver-core/src/test/unit/com/mongodb/internal/connection/DefaultConnectionPoolSpecification.groovy

```diff
@@ -466,6 +466,19 @@ class DefaultConnectionPoolSpecification extends Specification {
         thrown(MongoWaitQueueFullException)
     }
 
+    def 'invalidate should do nothing when pool is closed'() {
```

Comment:
Added a test for the pool as well, just to make sure we don't change the behavior in the future without thinking about it.